vars/infra.groovy

```diff
@@ -272,23 +272,16 @@ void prepareToPublishIncrementals() {
 void maybePublishIncrementals() {
     if (new InfraConfig(env).isRunningOnJenkinsInfra() && currentBuild.currentResult == 'SUCCESS') {
         stage('Deploy') {
-          timeout(300) {
-            node('maven || linux || windows') {
-                timeout(15) {
-                    withCredentials([string(credentialsId: 'incrementals-publisher-token', variable: 'FUNCTION_TOKEN')]) {
-                        if (isUnix()) {
-                            sh '''
-    curl --retry 10 --retry-delay 10 -i -H "Authorization: Bearer $FUNCTION_TOKEN" -H 'Content-Type: application/json' -d '{"build_url":"'$BUILD_URL'"}' "https://incrementals.jenkins.io/" || echo 'Problem calling Incrementals deployment function'
-                            '''
-                        } else {
-                            bat '''
-    curl.exe --retry 10 --retry-delay 10 -i -H "Authorization: Bearer %FUNCTION_TOKEN%" -H "Content-Type: application/json" -d "{""build_url"":""%BUILD_URL%""}" "https://incrementals.jenkins.io/" || echo Problem calling Incrementals deployment function
-                            '''
-                        }
-                    }
-                }
+            withCredentials([string(credentialsId: 'incrementals-publisher-token', variable: 'FUNCTION_TOKEN')]) {
+                httpRequest url: 'https://incrementals.jenkins.io/',
+                    httpMode: 'POST',
+                    contentType: 'APPLICATION_JSON',
+                    validResponseCodes: '100:599',
+                    timeout: 300,
+                    requestBody: /{"build_url":"$BUILD_URL"}/,
+                    customHeaders: [[name: 'Authorization', value: 'Bearer $FUNCTION_TOKEN']],
```

Comment:
```suggestion
                    customHeaders: [[name: 'Authorization', value: "Bearer $FUNCTION_TOKEN"]],
```

If the plugin interpolates env vars in supplied arguments, that is a [bug in the plugin](https://www.jenkins.io/doc/developer/plugin-development/pipeline-integration/#variable-substitutions) and we should not rely on the bad behavior.

Reply:
no it wasn't working thanks for this